service/history/chasm/components/activity/handler.go

```diff
@@ -0,0 +1,160 @@
+package activity
+
+import (
+	"context"
+	"time"
+
+	"go.temporal.io/server/service/history/chasm"
+)
+
+// This will be nexus
+type ActivityHandler struct {
+}
+
+type NewActivityRequest struct {
+	Input []byte
+
+	notifier EventNotifier
+}
+
+type NewActivityResponse struct {
+	RefToken []byte
+}
+
+type ScheduleRequest struct {
+	Input []byte
+}
+type ScheduleResponse struct{}
+
+type RecordStartedRequest struct {
+	RefToken []byte
+}
+
+type RecordStartedResponse struct {
+	RefToken []byte
+	Input    []byte
+}
+
+type RecordCompletedRequest struct {
+	RefToken []byte
+	Output   []byte
+}
+
+type RecordCompletedResponse struct{}
+
+type DescribeActivityRequest struct{}
+
+type DescribeActivityResponse struct {
+	IsAbandonded  bool
+	StartedTime   time.Time
+	CompletedTime time.Time
+}
+
+func (h *ActivityHandler) NewActivity(
+	ctx context.Context,
+	request *NewActivityRequest,
+) (*NewActivityResponse, error) {
+	resp, activityRef, err := chasm.NewEntity(
+		ctx,
+		chasm.EntityKey{
+			NamespaceID: "default",
+			BusinessID:  "memo",
+			// in V1 we probably don't support specifying instanceID,
+			// need to change persistence implementation for supporting that.
+			// InstanceID:  uuid.New().String(),
+		},
+		NewScheduledActivity,
+		request,
+		chasm.WithBusinessIDPolicy(
+			chasm.BusinessIDReusePolicyAllowDuplicate,
+			chasm.BusinessIDConflictPolicyFail,
+		),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	resp.RefToken = activityRef.Serialize()
+	return resp, err
+}
+
+func (h *ActivityHandler) RecordStarted(
+	ctx context.Context,
+	request *RecordStartedRequest,
+) (*RecordStartedResponse, error) {
+	// resp := &RecordStartedResponse{}
+
+	ref, err := chasm.DeserializeComponentRef(request.RefToken)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, startedActivityRef, err := chasm.UpdateComponent(
```

Comment:
As discussed yesterday, I wish we could get the ref in the update function and keep the ref out of this signature. This is good to start with though.